Public/Src/Cache/ContentStore/Library/Utils/BandwidthChecker.cs

```diff
@@ -45,93 +45,76 @@ public async Task<CopyFileResult> CheckBandwidthAtIntervalAsync(OperationContext
 
-                        copyCompleted = firstCompletedTask == copyTask;
-                        if (copyCompleted)
-                        {
-                            return await copyTask;
-                        }
-                        else if (context.Token.IsCancellationRequested)
-                        {
-                            context.Token.ThrowIfCancellationRequested();
-                        }
+                        return (result, bytesCopied);
+                    }
+                    else if (context.Token.IsCancellationRequested)
+                    {
+                        context.Token.ThrowIfCancellationRequested();
+                    }
 
-                        // Copy is not completed and operation has not been canceled, perform
-                        // bandwidth check 
-                        try
+                    // Copy is not completed and operation has not been canceled, perform
+                    // bandwidth check
+                    if (tryGetPosition(destinationStream, out var position))
+                    {
+                        var receivedMiB = (position - previousPosition) / BytesInMb;
+                        var currentSpeed = receivedMiB / _config.BandwidthCheckInterval.TotalSeconds;
+                        if (currentSpeed == 0 || currentSpeed < minimumSpeedInMbPerSec)
                         {
-                            var position = destinationStream.Position;
+                            // Ensure that we signal the copy to cancel
+                            copyCancellation.Cancel();
+                            traceCopyTaskFailures(copyTask);
 
-                            var receivedMiB = (position - previousPosition) / BytesInMb;
-                            var currentSpeed = receivedMiB / _config.BandwidthCheckInterval.TotalSeconds;
-                            if (currentSpeed == 0 || currentSpeed < minimumSpeedInMbPerSec)
-                            {
-                                return new CopyFileResult(CopyFileResult.ResultCode.CopyBandwidthTimeoutError, $"Average speed was {currentSpeed}MiB/s - under {minimumSpeedInMbPerSec}MiB/s requirement. Aborting copy with {position} copied");
-                            }
-
-                            previousPosition = position;
-                        }
-                        catch (ObjectDisposedException)
-                        {
-                            // If the check task races with the copy completing, it might attempt to check the position of a disposed stream.
-                            // Don't bother logging because the copy completed successfully.
+                            var result = new CopyFileResult(CopyFileResult.ResultCode.CopyBandwidthTimeoutError, $"Average speed was {currentSpeed}MiB/s - under {minimumSpeedInMbPerSec}MiB/s requirement. Aborting copy with {position} copied");

```

Comment:
and very nit again: I think we should change the message and actually use 'bytesCopied' there. Because position is the position and it may be not equal to 'bytes copied'.